test/unit/ngsiv2/amqpBinding-test.js

```diff
@@ -0,0 +1,320 @@
+
+            setTimeout(function() {
+                contextBrokerMock.done();
+                done();
+            }, 100);
+        });
+    });
+
+    describe('When a new multiple measure arrives to a Device routing key with a faulty payload', function() {
+        beforeEach(function() {
+            contextBrokerMock
+                .matchHeader('fiware-service', 'smartGondor')
+                .matchHeader('fiware-servicepath', '/gardens')
+                .post('/v2/entities/Second%20UL%20Device/attrs',
+                    utils.readExampleFile('./test/unit/ngsiv2/contextRequests/singleMeasure.json'))
+                .reply(204);
+        });
+
+        it('should silently ignore the error (without crashing)', function(done) {
+            channel.publish(config.amqp.exchange, '.1234.MQTT_2.attrs', new Buffer('notAULPayload '));
+
+            setTimeout(function() {
+                done();
+            }, 100);
+        });
+    });
+
+    describe('When single message with multiple measures arrive to a Device routing key', function() {
+        beforeEach(function() {
+            contextBrokerMock
+                .matchHeader('fiware-service', 'smartGondor')
+                .matchHeader('fiware-servicepath', '/gardens')
+                .post('/v2/entities/Second%20UL%20Device/attrs',
+                     utils.readExampleFile('./test/unit/ngsiv2/contextRequests/multipleMeasure.json'))
+                .reply(204);
+        });
+
+        it('should send one update context per measure group to the Contet Broker', function(done) {
+            channel.publish(config.amqp.exchange, '.1234.MQTT_2.attrs', new Buffer('temperature|23|humidity|98'));
+
+            setTimeout(function() {
+                contextBrokerMock.done();
+                done();
+            }, 100);
+        });
+    });
+
+    describe('When a message with multiple measure groups arrives to a Device routing key', function() {
+        beforeEach(function() {
+            contextBrokerMock
+                .matchHeader('fiware-service', 'smartGondor')
+                .matchHeader('fiware-servicepath', '/gardens')
+                .post('/v2/entities/Second%20UL%20Device/attrs',
+                    utils.readExampleFile('./test/unit/ngsiv2/contextRequests/singleMeasure.json'))
+                .reply(204);
+
+            contextBrokerMock
+                .matchHeader('fiware-service', 'smartGondor')
+                .matchHeader('fiware-servicepath', '/gardens')
+                .post('/v2/entities/Second%20UL%20Device/attrs',
+                    utils.readExampleFile('./test/unit/ngsiv2/contextRequests/secondSingleMeasure.json'))
+                .reply(204);
+        });
+
+        it('should send a two update context requests to the Context Broker one with each attribute', function(done) {
+            channel.publish(config.amqp.exchange, '.1234.MQTT_2.attrs', new Buffer('temperature|23#humidity|98'));
```

Comment:
I understand that the different in this case is that although both measures are sent using a single operation in the AMQP layer this causes two individual requests to CB, one per each measure. Is my interpretation correct?

Reply:
Ok, clear.

NTC